core/store/src/trie/mem/node/encoding.rs

```diff
@@ -236,8 +236,8 @@ impl MemTrieNodeId {
     }
 
     /// Increments the refcount, returning the new refcount.
-    pub(crate) fn add_ref(&self, arena: &mut Arena) -> u32 {
-        let mut ptr = self.as_ptr_mut(arena.memory_mut());
+    pub(crate) fn add_ref(&self, arena: &mut impl ArenaMemory) -> u32 {
```

Comment:
```suggestion
    pub(crate) fn add_ref(&self, memory: &mut impl ArenaMemory) -> u32 {
```

and similarly elsewhere.